mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -10010,8 +11199,12 @@ struct XdlopsGemmV2RewritePattern
       else
         sourceOffsetA.push_back(sourceOffsetBeforeTransformA);
 
-      auto valueA = lklb.create<memref::LoadOp>(loc, dataType, op.matrixA(),
-                                        ValueRange{sourceOffsetA});
+      Value valueA;
+      if (KPack > 1) {
+        valueA = emitLoadLogic(lklb, loc, op.matrixA().getType().template cast<MemRefType>(), argType, false, {}, op.matrixA(), ValueRange{sourceOffsetA});
```

Comment:
General comment: is it possible to statically compute the operations emitLoadLogic will lead to and apply those transformations to sourceOffsetA?

Reply:
the computation depends on `waveOffsetA`, which is derived from the current workitem ID, so it's not statically determined.